### PR TITLE
Fixup resource group

### PR DESCRIPTION
**Pull request by satanson**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #
https://github.com/StarRocks/starrocks/issues/4310
https://github.com/StarRocks/starrocks/issues/4334

### CODE REVIEW

**Review by laotan332 — DISMISSED** [review]

File: be/src/exec/workgroup/work_group.h [hunk 1]
```diff
@@ -229,6 +229,7 @@ class WorkGroupManager {
     std::shared_mutex _mutex;
     std::unordered_map<int128_t, WorkGroupPtr> _workgroups;
     std::unordered_map<int64_t, int64_t> _workgroup_versions;
+    std::unordered_map<std::string, int128_t> _unique_workgroup_names;
```
Comment (kangkaisen) [inline be/src/exec/workgroup/work_group.h:232]: why need this unique name map？I think the unique id could identify one work group.

**Comment by satanson** [comment]
run starrocks_fe_unittest

**Comment by satanson** [comment]
run starrocks_fe_unittest

**Comment by satanson** [comment]
@Mergify rebase

**Comment by satanson** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 7 / 17 (41.18%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/catalog/WorkGroupClassifier.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/WorkGroupAnalyzer.java|3|6|50.00%|
|:large_blue_circle:|com/starrocks/catalog/WorkGroupMgr.java|4|7|57.14%|

**Review by kangkaisen — DISMISSED** [review]

File: be/src/exec/workgroup/work_group.cpp [hunk 2]
```diff
@@ -194,6 +194,8 @@ void WorkGroupManager::apply(const std::vector<TWorkGroupOp>& ops) {
             _workgroups.erase(wg_it);
             _sum_cpu_limit -= wg_it->second->cpu_limit();
             _workgroup_expired_versions.erase(it++);
+            auto name = wg_it->second->name();
```
Comment (laotan332) [inline be/src/exec/workgroup/work_group.cpp:197]: What is the purpose of getting this local variable?

**Review by Seaven — DISMISSED** [review]

**Review by laotan332 — APPROVED** [review]

**Comment by satanson** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 7 / 17 (41.18%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/catalog/WorkGroupClassifier.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/WorkGroupAnalyzer.java|3|6|50.00%|
|:large_blue_circle:|com/starrocks/catalog/WorkGroupMgr.java|4|7|57.14%|

**Review by Seaven — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**